csrc/multidevice/communication.h

```diff
@@ -141,14 +140,23 @@ class Communication : public Expr {
   // the constructor that takes IrCloner aren't needed.
   bool sameAs(const Statement* other) const override;
 
-  // TODO: const CommParams&.
-  auto params() const {
+  const CommParams& params() const {
     return params_;
   }
 
+  bool isRootInMesh() const {
+    return params_.mesh.has(params_.root);
+  }
+
+  const Team& team();
+
  private:
-  // store the arguments of the communication
+  // Stores the arguments used to construct the communication.
   CommParams params_;
+  // This can be computed from params_, but given how frequently this is used
+  // in the hot path, I'm currently storing it as a field that'll be computed by
+  // Communication::team().
+  Team team_;
```

Comment:
Why not keep team creation in `lower_communication.cpp`? This will only cache team creation in the same `Communication`?

Reply:
Are you proposing to let lower_communication construct a Communication using `root`, `mesh` and `team`? I wanted to avoid that because `mesh` and `team` overlap quite a lot, so I chose to let lower_communication construct a Communication using `root` and `mesh` and let Communication compute `team` on the fly and cached.